pyro/distributions/kl.py

```diff
@@ -43,4 +43,16 @@ def _kl_independent_mvn(p, q):
     raise NotImplementedError
 
 
+# TODO: move upstream
+@register_kl(TransformedDistribution, TransformedDistribution)
+def _kl_transformed_transformed(p, q):
+    if p.transforms != q.transforms:
+        raise NotImplementedError
+    if p.event_shape != q.event_shape:
+        raise NotImplementedError
+    extra_event_dim = len(p.event_shape) - len(p.base_dist.event_shape)
```

Comment:
Should this be `len(p.batch_shape) - len(p.base_dist.batch_shape)`? I seem to remember a `Transform` that changed `event_dim`, which I believe would introduce skew here.

Reply:
Yeah, LKJ transforms a vector into a matrix. It is a bit subtle. I'll think more about this.

Reply:
Nice observation! Currently, transformed distributions assume that [`p.batch_shape + p.event_shape == p.base_dist.batch_shape + p.base_dist.event_shape`](https://github.com/pytorch/pytorch/blob/master/torch/distributions/transformed_distribution.py#L56). So I think that we can't construct a transformed distribution using a Transform that increases `event_dim`. Assuming that that issue is fixed, then using your formula would be better. I'll make the change. Thanks!

Reply:
Good observation, we should fix that upstream!

Reply:
It would be tricky. We'll need to provide more information to Transform class, e.g. to deal with the case:
```
d = dist.TransformedDistribution(dist.Normal(0, 1).expand((3,)), transforms.StickBreakingTransform())
d.event_shape == d.sample().shape  # 3 != 4
```
I think that users are less likely to use those transforms to create a transformed distribution. So I would prefer to skip it for now. :D